ravenframework/Optimizers/fitness/fitness.py

```diff
@@ -16,156 +16,166 @@
+  @ In, rlz, xr.Dataset, containing the evaluation of a set of individuals
+  @ In, kwargs, dict, dictionary of parameters:
+        objVar, list of strings or single string, name(s) of the objective variable(s)
+        scale, list of floats, scaling coefficient(s) for the objective function (default = 1.0 for each objective)
+        shift, list of floats, coefficient(s) for shifting the objective value (default = 0.0 for each objective)
+        constraintFunction, xr.DataArray, measuring the severity of the constraint violation
+        penalty, list of floats, penalties for constraint violations (default = 1.0 for each objective)
+        type, list of strings, indicating 'min' or 'max' for each objective
+  @ Out, fitnessSet, xr.Dataset, the fitness function for the given population.
   """
-  if kwargs['a'] == None:
-    a = 1.0
-  else:
-    a = kwargs['a']
-
-  if kwargs['b'] == None:
-    b = 0.0
-  else:
-    b = kwargs['b']
-
   objVar = kwargs['objVar']
-  val = rlz[objVar]
-  data = np.atleast_1d(rlz[objVar].data)
-  denom = 1.0 + np.exp(-a * (val - b))
-  fitness = 1.0 / denom
-  fitness = xr.DataArray(np.array(fitness),
-                          dims=['chromosome'],
-                          coords={'chromosome': np.arange(len(data))})
-
-  return fitness
-
+  scale = kwargs.get('scale', [1.0] * len(objVar))  # Scaling factors for objectives
+  shift = kwargs.get('shift', [0.0] * len(objVar))  # Shifting value for each objective
+  penalty = kwargs.get('penalty', [1.0] * len(objVar))  # Penalty for constraint violations
+  g = kwargs.get('constraintFunction', None)  # Constraint evaluations (if any)
+  fitnessSet = xr.Dataset()
+  for i, obj in enumerate(objVar):
+      data = np.atleast_1d(rlz[obj].data)  # Objective values
+      fitness = np.zeros(data.shape)
+      for ind in range(data.size):
+          # Base logistic fitness calculation
+          denom = 1.0 + np.exp(-scale[i] * (data[ind] - shift[i]))
+          fit = 1.0 / denom
+          # Apply penalties for constraint violations, if any
+          if g is not None and np.any(g.data[ind, :] < 0):  # Constraint violation
+              for constInd in range(g.data.shape[1]):
+                  fit -= penalty[i] * max(0, -g.data[ind, constInd])
+          # Adjust for maximization problems by negating the fitness value
+          if kwargs['type'][i] == 'max':
+              fit = 1.0 - fit  # Adjust the logistic fitness for maximization
+          fitness[ind] = fit
+      # Store fitness in the dataset
+      fitnessSet[obj] = xr.DataArray(fitness, dims=['chromosome'], coords={'chromosome': np.arange(len(data))})
+  return fitnessSet
 
 __fitness = {}
 __fitness['invLinear'] = invLinear
 __fitness['logistic']  = logistic
 __fitness['feasibleFirst'] = feasibleFirst
+#NOTE hardConstraint method will be used later once constraintHandling is realized. Until then, it will be commented. @JunyungKim
+# __fitness['hardConstraint'] = hardConstraint
```

Comment:
These lines can be removed, it seems the method is not implemented yet.

Reply:
@JunyungKim or @Jimmy-INL Are we planning on adding hardConstraints in a followup pull request?

Reply:
remove?

Reply:
Removed: f3e5e69d806d950b3e4e9b78672e03ab0d6e3d67